### DISCUSSION TITLE
Is it possible for Toolpad to enable the creation of a Single Page Application (SPA) without the need for a Node server?

### DISCUSSION THREAD

**Discussion by voquockhanh** [mui/toolpad#3071] [Help and support] [answered]
Apologies, as I have not yet thoroughly examined Toolpad's source code and am uncertain about which features necessitate server-side functionality. Assuming there are no indispensable features, or we have the option to disable them, could we consider facilitating the construction of Toolpad as a straightforward SPA? This approach could potentially lead to more cost-effective hosting solutions for our applications.

**Comment by Janpot** [comment, marked as answer]
Toolpad requires a server for its data fetching capabilities such as
* [HTTP requests](https://mui.com/toolpad/concepts/http-requests/): They are running from the server
* [functions](https://mui.com/toolpad/concepts/custom-functions/): Toolpad pages load backend data by calling server functions
* [data providers](https://mui.com/toolpad/concepts/data-providers/): Toolpad pages can load backend data into data grids through our data provider interface.

**Reply by voquockhanh** [reply]
Thank you Janpot for the prompt answer.
